### ISSUE TITLE
Unsupported parameters for (awx.awx.export) module: credentials, job_templates

### ISSUE THREAD

**Issue by itinneed2022** [ansible/awx#15345]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.
- [X] I am **NOT** reporting a (potential) security vulnerability. (These should be emailed to `security@ansible.com` instead.)

### Bug Summary

awx.awx.export returns:
TASK [Export workflow job template] ******************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Unsupported parameters for (awx.awx.export) module: credentials, job_templates. Supported parameters include: controller_password (tower_password), request_timeout, controller_host (tower_host), controller_oauthtoken (tower_oauthtoken), all, validate_certs (tower_verify_ssl), controller_username (tower_username), controller_config_file (tower_config_file)."}

The job_templates option should be available according to:
https://galaxy.ansible.com/ui/repo/published/awx/awx/content/module/export/?extIdCarryOver=true&sc_cid=701f2000001OH7YAAW

### AWX version

awxkit             24.6.1

### Select the relevant components

- [ ] UI
- [ ] UI (tech preview)
- [ ] API
- [ ] Docs
- [X] Collection
- [ ] CLI
- [ ] Other

### Installation method

N/A

### Modifications

no

### Ansible version

2.13.2

### Operating system

MacOS Sonoma 14.5

### Web browser

Firefox

### Steps to reproduce

ansible-playbook /Users/myself/Documents/Ansible/AnsibleAutomation_platform_rebuild/export_workflow.yaml

### Expected results

```
PLAY RECAP *******************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0 
```

### Actual results

```
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [Export Workflow] *******************************************************************************************************************************************************************************************

TASK [Export workflow job template] ******************************************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Unsupported parameters for (awx.awx.export) module: credentials, job_templates. Supported parameters include: controller_password (tower_password), request_timeout, controller_host (tower_host), controller_oauthtoken (tower_oauthtoken), all, validate_certs (tower_verify_ssl), controller_username (tower_username), controller_config_file (tower_config_file)."}

PLAY RECAP *******************************************************************************************************************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
```

### Additional information

```
python3 -m pip install awxkit
Requirement already satisfied: awxkit in /opt/homebrew/lib/python3.10/site-packages (24.6.1)
Requirement already satisfied: PyYAML in /opt/homebrew/lib/python3.10/site-packages (from awxkit) (6.0.1)
Requirement already satisfied: setuptools in /opt/homebrew/lib/python3.10/site-packages (from awxkit) (59.6.0)
Requirement already satisfied: requests in /opt/homebrew/lib/python3.10/site-packages (from awxkit) (2.32.3)
Requirement already satisfied: charset-normalizer<4,>=2 in /opt/homebrew/lib/python3.10/site-packages (from requests->awxkit) (3.3.2)
Requirement already satisfied: urllib3<3,>=1.21.1 in /opt/homebrew/lib/python3.10/site-packages (from requests->awxkit) (2.2.2)
Requirement already satisfied: idna<4,>=2.5 in /opt/homebrew/lib/python3.10/site-packages (from requests->awxkit) (3.7)
Requirement already satisfied: certifi>=2017.4.17 in /opt/homebrew/lib/python3.10/site-packages (from requests->awxkit) (2024.7.4)

```

**Comment by itinneed2022** [comment]
Also the playbook:
```
---
- name: Export Workflow
  hosts: localhost
  connection: local
  gather_facts: false
  collections:
    - awx.awx
  environment:
     CONTROLLER_HOST: https://ansibleautomation.mycompany.com/
     CONTROLLER_USERNAME: me2500
     CONTROLLER_PASSWORD: redacted
     CONTROLLER_VERIFY_SSL: False
  tasks:
    - name: Export workflow job template
      awx.awx.export:
        job_templates: "Server Decom - 1"
        credentials: 'all'
      register: export_results
    - debug:
        var: export_results
    - name: Export workflow job template to file
      copy:
        content: "{{ export_results | to_nice_yaml( width=50, explicit_start=True, explicit_end=True) }}"
        dest: workflow.yaml
```

**Comment by thedoubl3j** [comment]
thanks @itinneed2022 for reporting this, we will see if we can get a reproducer. can you confirm if the user you are running this as has access to all the secrets/credentials?

**Comment by fosterseth** [comment]
if you have a different way of detecting when nodes are down permanently, you might be able to log into a running task container and run the `awx-manage deprovision_instance` command to delete it from the database and those entries should go away

**Comment by fosterseth** [comment]
hmm I wasn't able to reproduce this locally. Can you verify the version of awxkit and awx collection you are using?

**Comment by itinneed2022** [comment]
> thanks @itinneed2022 for reporting this, we will see if we can get a reproducer. can you confirm if the user you are running this as has access to all the secrets/credentials?

The only credentials I'm using are the ones defined in the playbook.



> hmm I wasn't able to reproduce this locally. Can you verify the version of awxkit and awx collection you are using?

```
awx --version
24.6.1
```
How do I verify awxkit?

**Comment by djyasin** [comment]
Hello @itinneed2022,
Since we were unable to reproduce this locally, it seems that it may be an environmental issue. We would recommend reaching out on the [Ansible Community Forum](https://forum.ansible.com/tag/awx) to see if anyone else in the community as experienced this, and if so, how they were able to resolve it.

**Comment by itinneed2022** [comment]
I did a complete reinstall. awx, awkit, etc. That seems to have fixed it, only now I get this error:

```
fatal: [localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "",
    "module_stdout": "\n{\"failed\": true, \"msg\": \"Failed to export assets 'NoneType' object is not subscriptable\", \"exception\": \"  File \\\"/var/folders/_w/_1xw1qh52410qhfh4z08gc0h0000gq/T/ansible_awx.awx.export_payload_qhnq_ew6/ansible_awx.awx.export_payload.zip/ansible_collections/awx/awx/plugins/modules/export.py\\\", line 184, in main\\n  File \\\"/opt/homebrew/lib/python3.9/site-packages/awxkit/api/pages/api.py\\\", line 248, in export_assets\\n    endpoint = self._filtered_list(endpoint, value)\\n  File \\\"/opt/homebrew/lib/python3.9/site-packages/awxkit/api/pages/api.py\\\", line 227, in _filtered_list\\n    identifier = next(field for field in options['search_fields'] if field in ('name', 'username', 'hostname'))\\n\", \"invocation\": {\"module_args\": {\"workflow_job_templates\": [\"Server Decom - 1. SNOW into SPOG Workflow\"], \"credentials\": [\"all\"], \"controller_host\": \"https://my.company.com/\", \"controller_username\": \"me\", \"controller_password\": \"VALUE_SPECIFIED_IN_NO_LOG_PARAMETER\", \"validate_certs\": false, \"all\": false, \"request_timeout\": null, \"controller_oauthtoken\": null, \"controller_config_file\": null, \"users\": null, \"organizations\": null, \"teams\": null, \"credential_types\": null, \"notification_templates\": null, \"projects\": null, \"inventory\": null, \"inventory_sources\": null, \"job_templates\": null, \"execution_environments\": null, \"applications\": null, \"schedules\": null}}}\n\n{\"failed\": true, \"msg\": \"This endpoint raised an error: /api/v2/credentials/\\nThis endpoint raised an error: /api/v2/workflow_job_templates/\\n\", \"exception\": \"  File \\\"/var/folders/_w/_1xw1qh52410qhfh4z08gc0h0000gq/T/ansible_awx.awx.export_payload_qhnq_ew6/ansible_awx.awx.export_payload.zip/ansible_collections/awx/awx/plugins/modules/export.py\\\", line 187, in main\\n  File \\\"/var/folders/_w/_1xw1qh52410qhfh4z08gc0h0000gq/T/ansible_awx.awx.export_payload_qhnq_ew6/ansible_awx.awx.export_payload.zip/ansible_collections/awx/awx/plugins/module_utils/controller_api.py\\\", line 278, in fail_json\\n    super().fail_json(**kwargs)\\n  File \\\"/var/folders/_w/_1xw1qh52410qhfh4z08gc0h0000gq/T/ansible_awx.awx.export_payload_qhnq_ew6/ansible_awx.awx.export_payload.zip/ansible/module_utils/basic.py\\\", line 1554, in fail_json\\n    sys.exit(1)\\n\", \"invocation\": {\"module_args\": {\"workflow_job_templates\": [\"Server Decom - 1. SNOW into SPOG Workflow\"], \"credentials\": [\"all\"], \"controller_host\": \"https://my.company.com/\", \"controller_username\": \"me\", \"controller_password\": \"VALUE_SPECIFIED_IN_NO_LOG_PARAMETER\", \"validate_certs\": false, \"all\": false, \"request_timeout\": null, \"controller_oauthtoken\": null, \"controller_config_file\": null, \"users\": null, \"organizations\": null, \"teams\": null, \"credential_types\": null, \"notification_templates\": null, \"projects\": null, \"inventory\": null, \"inventory_sources\": null, \"job_templates\": null, \"execution_environments\": null, \"applications\": null, \"schedules\": null}}}\n",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
```

**Comment by itinneed2022** [comment]
That error is due to the user I was given requiring an sso login. I created a non-SSO user and now I can access the api's.

**Comment by Klaas-** [comment]
I think the error here is just misleading:
`[fatal: [localhost]: FAILED! => {"changed": false, "msg": "Unsupported parameters for (awx.awx.export) module: credentials, job_templates. Supported parameters include: controller_password (tower_password), request_timeout, controller_host (tower_host), controller_oauthtoken (tower_oauthtoken), all, validate_certs (tower_verify_ssl), controller_username (tower_username), controller_config_file (tower_config_file)."}`
is (or whatever the parameter is you had specified) is what is happening when awxkit is not installed in your python env

**Comment by GitarPlayer** [comment]
The error is misleading indeed. If you have awxkit installed but did not source your venv it also pops up.

**Comment by williamhanseneric** [comment]
I struggled with this same problem with a controller node.  I tried both an activated 3.9.12 python venv and a python venv running the system 3.6.8 with different versions of ansible and awxkit.  I finally fell back to a python venv that I had used a couple years ago, but was still not not successful,

Still ended up with errors,
Unsupported parameters for (ansible.controller.export) module: job_templates. Supported parameters include: all, controller_config_file, controller_host, controller_oauthtoken, controller_password, controller_username, request_timeout, validate_certs (tower_config_file, tower_host, tower_oauthtoken, tower_password, tower_username, tower_verify_ssl).

The trick was to also set 'ansible_python_interpreter' to the full internal path to python for this ansible 2.10.7 venv I was using a couple of years ago within the playbook as a fact.  Without doing this, *even when running the playbook underneath that activated python venv*...  I would still get errors.

Maybe this helps someone, it sure is a royal pain to deal with these things.  My recollection from the past is that I was never able to get this code working above this ansible 2.10.7 venv in the past.  My guess was some internal conflict with the AAP install.
